pysindy/differentiation/finite_difference.py

```diff
@@ -63,6 +82,8 @@ def _forward_difference(self, x, t=1):
         Note that in order to maintain compatibility with sklearn the,
         array returned, x_dot, always satisfies np.ndim(x_dot) == 2.
         """
+        if self.is_uniform and not np.isscalar(t):
```

Comment:
Do we do any validation of the inputs, to make sure the grid passed in actually is uniform? If not, should we do that?

Reply:
It would be good, but I wasn't sure the best way to do this. Seems like would need to loop through the x array, which would slow down the differentiation significantly.